superduperdb/ext/torch/model.py

```diff
@@ -94,45 +93,152 @@ def __post_init__(self):
             self.target_preprocessors = Artifact(artifact=self.target_preprocessors)
 
 
-@dc.dataclass
-class Base:
-    collate_fn: t.Optional[t.Union[Artifact, t.Callable]] = None
+@dc.dataclass(kw_only=True)
+class TorchModel(Model):
     is_batch: t.Optional[t.Union[Artifact, t.Callable]] = None
     num_directions: int = 2
-    metrics: t.Optional[t.Sequence[t.Union[str, Metric]]] = None
-    training_select: t.Optional[Select] = None
+    optimizer_state: t.Optional[Artifact] = None
+    forward_method: str = '__call__'
+    train_forward_method: str = '__call__'
+
+    def __post_init__(self):
+        super().__post_init__()
+
+        if self.model_to_device_method:
+            s.logging.debug(
+                f'{self.model_to_device_method} will be overriden with `to`'
+            )
+
+        self.model_to_device_method = 'to'
+
+        self.object.serializer = 'torch'
+
+        if self.optimizer_state is not None:
+            self.optimizer.load_state_dict(self.optimizer_state.artifact)
+
+        self._validation_set_cache = {}
+
+    def to(self, device):
+        self.object.artifact.to(device)
+
+    @cached_property
+    def optimizer(self):
+        return self.training_configuration.optimizer_cls.artifact(
+            self.object.artifact.parameters(),
+            **self.training_configuration.optimizer_kwargs,
+        )
+
+    @property
+    def optimizers(self) -> t.List:
+        return [self.optimizer]
```

Comment:
return (self.optimizer) 
if self.optimizers are meant to be used as immutable

Reply:
This is from ages ago, not sure why this is in the `diff`.